server/src/main/resources/UI_Interaction.puml

```diff
@@ -0,0 +1,54 @@
+@startuml
+participant TerraUI
+box "Leo VM"
+    participant JupyterExtension
```

Comment:
What is JupyterExtension vs JupyterAutoSaveExtension? Maybe Jupyter UI Extension and Jupyter Server Extension would be clearer (I assume that is the distinction here)